ext4:  check return of ext4_orphan_get properly

From: Josef Bacik <jbacik@redhat.com>

This patch fix a panic while running fsfuzzer. 
We are improperly checking the return of ext4_orphan_get.

Signed-off-by: Josef Bacik <jbacik@redhat.com>
Signed-off-by: Mingming Cao <cmm@us.ibm.com>


---
 fs/ext4/super.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

Index: linux-2.6.25-rc9/fs/ext4/super.c
===================================================================
--- linux-2.6.25-rc9.orig/fs/ext4/super.c	2008-04-14 11:11:34.000000000 -0700
+++ linux-2.6.25-rc9/fs/ext4/super.c	2008-04-14 11:11:48.000000000 -0700
@@ -1593,8 +1593,8 @@ static void ext4_orphan_cleanup (struct 
 	while (es->s_last_orphan) {
 		struct inode *inode;
 
-		if (!(inode =
-		      ext4_orphan_get(sb, le32_to_cpu(es->s_last_orphan)))) {
+		inode = ext4_orphan_get(sb, le32_to_cpu(es->s_last_orphan));
+		if (IS_ERR(inode)) {
 			es->s_last_orphan = 0;
 			break;
 		}
